histomicstk/cli/utils.py

```diff
@@ -65,10 +66,21 @@ def segment_wsi_foreground_at_low_res(ts, lres_size=2048):
 
     im_lres, _ = ts.getRegion(
         scale=fgnd_seg_scale,
-        format=large_image.tilesource.TILE_FORMAT_NUMPY
+        format=large_image.tilesource.TILE_FORMAT_NUMPY,
+        frame=frame
     )
 
-    im_lres = im_lres[:, :, :3]
+    # check number of channels
+    if len(im_lres.shape) <= 2 or im_lres.shape[2] == 1:
+        im_lres = np.dstack((im_lres, im_lres, im_lres))
+        if default_img_inversion:
+            invert_image = True
+    else:
+        im_lres = im_lres[:, :, :3]
+
+    # perform image inversion
+    if invert_image:
+        im_lres = np.max(im_lres) - im_lres
```

Comment:
At least some of the issues we've seen is because, when we composite images using a style parameter, it can return the image as a numpy array with a dtype of `float64` even though the data is intrinsically `uint8`.  `im_lres` in this case is `float64`, and the `simple_mask` function calls `skimage.rgb2gray`, which, if the data type is float, assumes the values are scaled from [0,1] and produces erroneous results.

Adding something like:
```
    if issubclass(im_lres.dtype.type, np.floating) and np.max(im_lres) > 1:
        if np.min(im_lres) >= 0 and np.max(im_lres) < 256:
            im_lres = im_lres.astype(np.uint8)
        if np.min(im_lres) >= 0 and np.max(im_lres) < 65536:
            im_lres = im_lres.astype(np.uint16)
```
would fix this for the conditions we care about.

Reply:
I tried a similar strategy yesterday. There was some warning that the `np.exp()` was overloaded. I guess it was because we were sending float64 values to it and it had to do a lot of calculations to give the output. So I tried casting to `im_lms = np.exp(im_lms.astype(np.uint8))`. It was able to detect nuclei but was giving erroneous results. This looks like a better way to cast even before it goes to the exponential function.